src/plugins/preauth/pkinit/pkinit_crypto_openssl.c

```diff
@@ -4636,6 +4636,42 @@ reassemble_pkcs11_name(pkinit_identity_opts *idopts)
     return ret;
 }
 
+static int
+hex_string_to_bin(const char *str, int *bin_len_out, CK_BYTE **bin_out)
+{
+    size_t str_len, i;
+    CK_BYTE *bin;
+    char *endptr, tmp[3] = { '\0', '\0', '\0' };
+    long val;
+
+    *bin_out = NULL;
+
+    str_len = strlen(str);
+    if (str_len % 2 != 0)
+        return EINVAL;
+    bin = malloc(str_len / 2);
+    if (bin == NULL)
+        return ENOMEM;
+
+    errno = 0;
+    for (i = 0; i < str_len / 2; i += 2) {
```

Comment:
I think the loop update should be `i++` here?

Reply:
Whoops, that would have been bad.  To mitigate the risk of having introduced other bugs, I temporarily brought back Sumit's tests after fixing this (and the bin_len_out initialization) and made sure they pass.